services/cli-host/automatos_cli_host/api.py

```
@@ -0,0 +1,94 @@
+    def _request(self, method: str, path: str, body: Optional[Dict[str, Any]] = None,
+                 *, auth: bool = True) -> Dict[str, Any]:
+        url = f"{self.base_url}{path}"
+        data = json.dumps(body).encode("utf-8") if body is not None else None
+        req = urllib.request.Request(url, data=data, method=method)
+        req.add_header("Accept", "application/json")
+        if data is not None:
+            req.add_header("Content-Type", "application/json")
+        if auth and self.token:
+            req.add_header(HOST_TOKEN_HEADER, self.token)
+        try:
+            with urllib.request.urlopen(req, timeout=self.timeout) as resp:
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)